app/src/test/mocks/mocks.ts

```diff
@@ -21,20 +19,21 @@ export function mockMatch<P>(params?: P) :match<P> {
     };
 }
 
-export function mockHistory(params?: any) {
-    return Object.assign({
+export function mockHistory(params?: Partial<History>): History {
+    const template: History = {
         length: 1,
-        action: "",
-        location: "",
+        action: "PUSH",
+        location: mockLocation(),
         push: () => {},
         replace: () => {},
         go: () => {},
         goBack: () => {},
         goForward: () => {},
-        block: () => {},
-        listen: () => {},
-        createHref: () => {}
-    }, params);
+        block: (() => {}) as any,
+        listen: (() => {}) as any,
+        createHref: () => ""
+    };
+    return Object.assign(template, params);
```

Comment:
The dead code inspector wanted to remove most of these properties because we weren't using them, so I made the method strongly typed.